lib/libvarnish/vcli_serve.c

```diff
@@ -428,6 +440,12 @@ VCLS_AddFd(struct VCLS *cs, int fdi, int fdo, cls_cb_f *closefunc, void *priv)
 	cfd->cli->vlu = VLU_New(cfd, cls_vlu, *cs->maxlen);
 	cfd->cli->sb = VSB_new_auto();
 	cfd->cli->limit = cs->limit;
+	if (cs->ws_sz > 0) {
+		cfd->cli->ws = malloc(sizeof(*cfd->cli->ws));
```

Comment:
It made things even more confusing :)

* `*heritage.param = mgt_param`
* `heritage.param = VSM_common_alloc(...)`
* `cache_param = heritage.param;`

Shouldn't it be passed as a pointer like `cli_buffer` and `cli_limit` to capture changes over time?